sampletester/inputs.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import glob
+import itertools
+import logging
+import os
+from functools import reduce
+from typing import Set
+
+from sampletester import parser
+from sampletester.parser import SCHEMA_TYPE_ABSENT as UNKNOWN_TYPE
+from sampletester.sample_manifest import SCHEMA_TYPE_VALUE as MANIFEST_TYPE
+from sampletester.testplan import SCHEMA_TYPE_VALUE as TESTPLAN_TYPE
+
+
+def untyped_yaml_resolver(unknown_doc: parser.Document) -> str :
+  """Determines how `parser.IndexedDocs` should classify `unknown_doc`
+
+  This is a resolver for parser.IndexedDocs, used to resolve YAML docs that did
+  not have a type field and thus could not be automatically classified. This
+  resolver resolves using the filename, for backward compatibility: files ending
+  in `.manifest.yaml` are categorized as manifest files, and remaining YAML
+  files are categorized as testplan files.
+  """
+  ext_split = os.path.splitext(unknown_doc.path)
+  ext = ext_split[-1]
+  if ext == ".yaml":
+    prev_ext = os.path.splitext(ext_split[0])[-1]
+    if prev_ext == ".manifest":
+      return MANIFEST_TYPE
+    else:
+      return TESTPLAN_TYPE
+  return UNKNOWN_TYPE
+
+
+def index_docs(*file_patterns: str) -> parser.IndexedDocs:
+  """Obtains manifests and testplans by indexing the specified paths or cwd.
+
+  This function works in the following sequence:
+
+  1. It attempts to obtain all the manifests and testplans contained in
+  the globs in `file_patterns`.
+
+  2. If either no manifest or no testplan is obtained this way:
+
+    2.1 if any of the globs in `file_patterns` resolved to a directory, it does
+        not search for any more files.
+
+    2.2 if none of the globs in `file_patterns` resolved to a directory, it
+        searches in all the paths under the cwd and registers the
+        `file_patterns` matching the types not found in the previous step. In
+        other words, if no manifests are found via the globs in `file_patterns`,
+        it attempts to get manifests under the cwd, and similarly for testplans.
+
+  Returns: the indexed docs of the files that were searched for.
+  """
+  explicit_paths = get_globbed(*file_patterns)
```

Comment:
It seems this line does `1. ` describes. Does that mean we can do an empty check here, and if `explicit_paths` is not empty we can return?

Reply:
No, because after we resolve the globs we need to
1. get the files under any of these resolved globs that turned out to be a directory
2. from all the files we get directly or indirectly from the resolved globs, we need to check that we get at least one testspec doc and at least one manifest doc.